### PR TITLE
refact db migration, add logger factory to ContainerConfig, add loggers

### PR DESCRIPTION
**Pull request by dtmypd**
to TodoService and UserService

### CODE REVIEW

File: src/App/Services/TodoService.php [hunk 1]
```diff
@@ -11,9 +11,18 @@
 use ExtendedSlim\Http\HttpCodeConstants;
 use ExtendedSlim\Http\RestApiResponse;
 use Exception;
+use Monolog\Logger;
 
 class TodoService
 {
+    const CREATE_LOG_MSG        = 'todo create';
```
[the remark this answers is not in the document]
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

File: src/App/Services/TodoService.php [hunk 2]
```diff
@@ -47,12 +57,19 @@ public function create(string $name, int $userId)
 
             $this->connection->commit();
 
+            $this->logger->info(self::CREATE_LOG_MSG, [
+                'name'      => $name,
+                'userId'    => $userId
```
[the remark this answers is not in the document]
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

File: src/App/Services/TodoService.php [hunk 3]
```diff
@@ -47,12 +57,19 @@ public function create(string $name, int $userId)
...
+            ]);
+
             return new RestApiResponse();
         }
         catch (Exception $e)
         {
             $this->connection->rollBack();
 
+            $this->logger->error(self::CREATE_ERROR_LOG_MSG, [ 'exception' => $e ]);
```
[the remark this answers is not in the document]
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

File: src/App/Services/TodoService.php [hunk 4]
```diff
@@ -111,10 +134,16 @@ public function getById(int $id): RestApiResponse
     {
         try
         {
-            return new RestApiResponse($this->todoRepository->getById($id));
+            $response = new RestApiResponse($this->todoRepository->getById($id));
+            $this->logger->info(self::BYID_LOG_MSG, [ 'id' => $id ]);
```
[the remark this answers is not in the document]
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

File: src/App/Services/TodoService.php [hunk 5]
```diff
@@ -111,10 +126,16 @@ public function getById(int $id): RestApiResponse
...
+            $this->logger->info('todo by id', ['id' => $id]);
+            return $response;
         }
         catch (RecordNotFoundException $e)
         {
+            $this->logger->error('todo by id error', [
+                'id'        => $id,
+                'exception' => $e
+            ]);
             return new RestApiResponse(
```
Comment (krizsanbalazs) [inline src/App/Services/TodoService.php]: Ide meg egy uj sort legyszi :)

File: src/App/Services/TodoService.php [hunk 6]
```diff
@@ -111,10 +126,16 @@ public function getById(int $id): RestApiResponse
     {
         try
         {
-            return new RestApiResponse($this->todoRepository->getById($id));
+            $response = new RestApiResponse($this->todoRepository->getById($id));
+            $this->logger->info('todo by id', ['id' => $id]);
+            return $response;
```
Comment (krizsanbalazs) [inline src/App/Services/TodoService.php]: Return eleg egy entert legyszi.

File: src/App/Services/TodoService.php (hunk 5 above)
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

File: src/App/Services/TodoService.php (hunk 6 above)
Comment (dtmypd) [reply src/App/Services/TodoService.php]: done

**Pull request merged**